src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -582,13 +583,49 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
       def processAccountsPage(
           taggedAccounts: List[BlockTagged[AccountsIndex]],
           taggedDelegateKeys: List[BlockTagged[DelegateKeys]]
-      ): Future[(Option[Int], Option[Int], List[BlockTagged[DelegateKeys]])] =
-        db.run(TezosDb.writeAccountsAndCheckpointDelegates(taggedAccounts, taggedDelegateKeys))
-          .map {
+      ): Future[(Option[Int], Option[Int], List[BlockTagged[DelegateKeys]])] = {
+        for {
+          activatedOperations <- fetchActivatedOperationsByLevel(taggedAccounts.map(_.blockLevel).distinct)
+          activatedAccounts <- db.run(TezosDb.findActivatedAccounts)
+          updatedTaggedAccounts = updateTaggedAccountsWithIsActivated(
+            taggedAccounts,
+            activatedOperations,
+            activatedAccounts.toList
+          )
+          res <- db.run(TezosDb.writeAccountsAndCheckpointDelegates(updatedTaggedAccounts, taggedDelegateKeys)).map {
             case (accountWrites, accountHistoryWrites, delegateCheckpoints) =>
               (accountWrites, delegateCheckpoints, taggedDelegateKeys)
           }
-          .andThen(logWriteFailure)
+        } yield res
+      }.andThen(logWriteFailure)
+
+      def updateTaggedAccountsWithIsActivated(
+          taggedAccounts: List[BlockTagged[AccountsIndex]],
+          activatedOperations: Map[Int, Seq[Tables.OperationsRow]],
+          activatedAccountIds: List[String]
+      ): List[BlockTagged[Map[AccountId, Account]]] =
+        taggedAccounts.map { taggedAccount =>
+          val activatedAccountsHashes = activatedOperations.get(taggedAccount.blockLevel).toSeq.flatten.map(_.pkh)
+          taggedAccount.copy(
+            content = taggedAccount.content.mapValues { account =>
+              val hash = account.manager.map(_.value)
+              if (activatedAccountsHashes.contains(hash) || activatedAccountIds.contains(hash.getOrElse(""))) {
```

Comment:
Can't we convert the activated container to only have one type of values, so that here we only make a single check? Each of these verifications might be scanning the whole sequence of activations...
I mean something similar to first lowercasing a string, instead of checking both for lower and upper case versions.

Moreso, if the accounts are not just a few, it might be useful to first prepare the operations-map into something directly useful to all checks, as a first step. E.g. make it a Map[Int, Set[Hash]] or similar.

Reply:
made a change to make one check on a single list